### ISSUE TITLE
Support for ESM added, then removed?

### ISSUE THREAD

**Issue by tannerstern** [davidmerfield/randomColor#160]
Hey @davidmerfield!

Thanks for this package, its a neat and simple tool that I find really useful, especially for seeding colors based on inputs.

I wanted to ask a question though: I see ESM support was added in #134 but then shortly after was removed in ff971faaa792736ab386d99c56127e2fa8da11ca. I'm wondering, was that on account of the changes from the PR breaking use of this module in other environments? That seems like what was going on. Since the version of this package available on `npm` is Node-first (non-browser compatible), it makes sense that going fully ESM would cause issues.

Maybe I've answered my own question, but at any rate, ESM-native imports would be a nice thing to have, especially through `npm`. Do you plan on re-adding ESM support in the future (of course, this being open-source I am not asking or demanding anything, just curious if that's a direction you are planning to go in).

Thanks!

**Comment by tannerstern** [comment]
That makes sense. After reading your message, I did a bit of digging to see how others have attempted to solve this issue (because there appear to be a lot of packages that have solved this, with build processes of varying complexity). I found a few things that were helpful in understanding what a solution _could_ look like:
- [How to Create a Hybrid NPM Module for ESM and CommonJS. | SenseDeep](https://www.sensedeep.com/blog/posts/2021/how-to-create-single-source-npm-module.html)
- [Ship ESM & CJS in one Package](https://antfu.me/posts/publish-esm-and-cjs)
- [Modules: Packages | Node.js v20.5.1 Documentation (Conditional Exports)](https://nodejs.org/api/packages.html#conditional-exports)

TL;DR Node has a way to provide support for both `require` and `import` with the `exports` field in `package.json` (docs in link 3). Per the "SenseDeep" example, one could:
1. Convert the codebase to ESM (@keithamus has already done this in [his fork](https://github.com/keithamus/randomColor/))
2. Add a minimal build step to yield two endpoints, one for CJS and one for ESM
3. Update `package.json` to point to the appropriate endpoint depending on the environment
4. (Bonus: as a fallback for those running on Node versions _before_ `exports`, the `main` field can be left in; `exports` takes precedence, but is ignored on older versions of Node)

I'm not sure what that build step would look like (Babel? Rollup? something else?) but it would surely be a simple configuration 👍
